lib-api/src/main/java/org/ergoplatform/appkit/BoxOperations.java

```diff
@@ -236,6 +269,9 @@ public UnsignedTransaction mintTokenToContractTxUnsigned(ErgoContract contract,
         if (!tokensToSpend.isEmpty()) {
             throw new IllegalArgumentException("Mint token not possible with spending tokens");
         }
+        if (attachment != null) {
+            throw new IllegalArgumentException("Mint token not possible with attachment");
```

Comment:
Minting a token sets the registers according to EIP-4 which can interphere with the attachment

Reply:
But can we extend EIP4 to allow attachments in the box with the minted asset?

Reply:
Sure, EIP4 could be extended to use attachments instead of its current way to store data. But when we do so, minting a token is still managed by EIP-4 and Eip4Token implementations, so this error would be still necessary.